whirlpool/auth.py

```diff
@@ -95,8 +95,11 @@ async def _do_auth(self, refresh_token):
                         return json.loads(await r.text())
                     elif refresh_token:
                         return await self._do_auth(refresh_token=None)
+                    elif r.status != 200:
+                        raise aiohttp.ClientConnectionError
```

Comment:
```suggestion
```

This is a real error, and we should not treat it as a timeout on HA. If there is an HTTP error coming from the server, HA should just show an error message, no retry.

Reply:
This error is one I added to my Aladdin Connect integration (copied from some other integration which I've forgotten at this point) - If the server complains with "not 200" should retry, say the server is busy, or some other complaint besides not authorized.  Otherwise the integration just hangs and the entities become "unavailable" and "no longer provided".
I suppose it could be some other locally defined error..